src/sources/splunk_hec.rs

```diff
@@ -30,12 +30,22 @@ lazy_static! {
 
 /// Accepts HTTP requests.
 #[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields, default)]
 pub struct SplunkConfig {
     /// Local address on which to listen
     #[serde(default = "default_socket_address")]
     address: SocketAddr,
     /// Splunk HEC token
-    token: String,
+    token: Option<String>,
+}
+
+impl Default for SplunkConfig {
```

Comment:
I don't think we need this + `serde(default)` since we already set a default for address and token has an Option which defaults to none.

Reply:
Compiler says differently

```
error[E0277]: the trait bound `sources::splunk_hec::SplunkConfig: std::default::Default` is not satisfied
  --> src/sources/splunk_hec.rs:32:10
   |
32 | #[derive(Deserialize, Serialize, Debug, Clone)]
   |          ^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `sources::splunk_hec::SplunkConfig`
   |
   = help: see issue #48214
```

I suppose `Deserialize` derive is somehow hitting [this issue](https://github.com/rust-lang/rust/issues/48214).

For note:
deriving `Default` doesn't work also

```
error[E0277]: the trait bound `std::net::SocketAddr: std::default::Default` is not satisfied
  --> src/sources/splunk_hec.rs:37:5
   |
37 |     address: SocketAddr,
   |     ^^^^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `std::net::SocketAddr`
   |
   = note: required by `std::default::Default::default`
```